UnityProject/Assets/Scripts/Tilemaps/Behaviours/Layers/MetaTileMap.cs

```diff
@@ -204,7 +204,12 @@ public void Update()
 
 		private void ApplyTileChange(TileLocation tileLocation)
 		{
-			if (mainThread.Equals(Thread.CurrentThread))
+			if (QueuedChanges.Contains(tileLocation))
+			{
+				return;
+			}
```

Comment:
What happens if a new change comes in before the old one is processed? Does this just stop the new one?

Reply:
actually nvm, the TileLocation would be different right?

Reply:
itd be the same, its a weird system